filament/src/details/IndexBuffer.cpp

```diff
@@ -64,6 +69,15 @@ IndexBuffer::Builder& IndexBuffer::Builder::name(utils::StaticString const& name
     return BuilderNameMixin::name(name);
 }
 
+IndexBuffer::Builder& IndexBuffer::Builder::async(backend::CallbackHandler* handler,
+        AsyncCallbackType callback, void* user) noexcept {
+    mImpl->mAsynchronous = true;
+    mImpl->mAsyncCreationHandler = handler;
+    mImpl->mAsyncCreationCallback = std::move(callback);
+    mImpl->mAsyncCreationUserData = user;
+    return *this;
+}
+
 IndexBuffer* IndexBuffer::Builder::build(Engine& engine) {
     return downcast(engine).createIndexBuffer(*this);
```

Comment:
This is where you should check `FILAMENT_CHECK` that async mode is enabled if `mAsynchronous` is `true`. All validation of the `Builder` should happen here.